tests_e2e/tests/lib/agent_test.py

```diff
@@ -59,8 +69,28 @@ def run_from_command_line(cls):
             cls(AgentTestContext.from_args()).run()
         except SystemExit:  # Bad arguments
             pass
+        except AssertionError as e:
+            log.error("%s", e)
+            sys.exit(1)
         except:  # pylint: disable=bare-except
             log.exception("Test failed")
             sys.exit(1)
 
         sys.exit(0)
+
+    def _run_remote_test(self, command: str, use_sudo: bool = False, attempts: int = ATTEMPTS, attempt_delay: int = ATTEMPT_DELAY) -> None:
+        """
+        Derived classes can use this method to execute a remote test (a test that runs over SSH).
+        """
+        try:
+            ssh_client: SshClient = self._context.create_ssh_client()
```

Comment:
How about initializing this to AgentTest instead every remote run in test_suite?

Reply:
sorry, i don't understand the question

Reply:
It seems we initialize ssh_client on every remote run, can we optimize this?

Reply:
initialization of the SSH client is very cheap (just 4 assigments). it was designed this way so that we can create instances at will, without worrying about it.